src/mailadm/app.py

```diff
@@ -3,22 +3,37 @@
 """
 import sys
 import os
-
+import deltachat
 from .web import create_app_from_db_path
 import time
 import threading
 from .db import get_db_path, DB
-from mailadm.commands import prune
+from mailadm.commands import prune, warn_expiring_users
 from mailadm.bot import main as run_bot
 from mailadm.bot import get_admbot_db_path
 
 
 def prune_loop():
     db = DB(get_db_path())
+    ac = deltachat.Account(get_admbot_db_path())
+    botconfigured = True
+    try:
+        ac.run_account()
+    except AssertionError:
+        botconfigured = False
+    # how to shut down admbot account in the end?
     while 1:
         for logmsg in prune(db).get("message"):
             # the file=sys.stderr seems to be necessary so the output is shown in `docker logs`
             print(logmsg, file=sys.stderr)
+        if not botconfigured:
+            try:
+                ac.run_account()
+                botconfigured = True
+            except AssertionError:
```

Comment:
not a good practise to catch AssertionError. Better make sure to throw at least a ValueError or a custom Exception and catch it here.

Reply:
obsolete because of this:

> We called and decided that we will not support the use case of running mailadm without a bot; the prune thread will only be started as soon as we have a working bot account object.

now I don't need that part anymore :)